devito/builtins/utils.py

```diff
@@ -131,3 +132,23 @@ def wrapper(*args, **kwargs):
         return func(*processed, argmap=argmap, **kwargs)
 
     return wrapper
+
+
+def check_args(func):
+    """
+    Perform checks on the arguments supplied to a builtin.
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        for i in args:
+            try:
```

Comment:
Should only be an issue on Device platform, CPU transient do hold the data

Reply:
OK, tweaking